lib/loggers/structured_logging.rb

```diff
@@ -0,0 +1,130 @@
+# frozen_string_literal: true
+
+require "active_support/core_ext/module/delegation"
+require "active_support/core_ext/object/blank"
+require "json"
+require "logger"
+require "active_support/logger"
+
+# Wraps any standard Logger object to provide structured logging capabilities.
+#
+# May be called with a block:
+#
+#   logger = StructuredLogging.new(Logger.new(STDOUT))
+#   logger.context(key: value) { logger.info 'Stuff' } # Logs "{ message: 'Stuff', key: 'value' }"
+module StructuredLogging
```

Comment:
This is really complicated. Could we put it in Koi instead of copying?

Reply:
Moved to Koi: https://github.com/katalyst/koi/pull/571